lib/asset.js

```diff
@@ -13,25 +13,36 @@
 'use strict';
 const path = require('path');
 
-class Source {
-    constructor(params, directory="", name="") {
-        this.settings = params;
-
+class Asset {
+    constructor(assetParams, directory="", name="") {
+        this.params = assetParams;
         this.name = name;
         this.path = path.join(directory, this.name);
     }
 
     get type() {
-        return this.settings.type;
+        if(this.params){
+            return this.params.type;
+        } else {
+            return undefined;
+        }
```

Comment:
This is a bit shorter:
```
return this.params && this.params.type;
```

I learned that you basically never have to use/return `undefined` explicitly in js.